src/processor/result/factorized_table.cpp

```diff
@@ -144,15 +144,18 @@ void FactorizedTable::lookup(std::vector<ValueVector*>& vectors,
     uint64_t numTuplesToRead) const {
     assert(vectors.size() == colIdxesToScan.size());
     for (auto i = 0u; i < colIdxesToScan.size(); i++) {
+        auto vector = vectors[i];
+        // TODO(Xiyang/Ziyi): we should set up a rule about when to reset. Should it be in operator?
+        vector->resetAuxiliaryBuffer();
```

Comment:
Maybe the source operator of each pipeline should be responsible for resetting the vector state?
We may need an api in vector which resets the current state of the vector.